### PR TITLE
Fix: Bug 2122409 - pki-tomcat/kra unable to decrypt when using RSA-OAEP padding in RHEL9 with FIPS enabled

### PR DESCRIPTION
**Pull request by fmarco76**
The purpose of this patch is to continue the improvement of this bug in 2 ways:

1. Create a pikspawn variable to cause pkispawn to create a subsystem configured for oaep `pki_use_oaep_rsa_keywrap=True`, the default is `False`.

2. Improve the rest calls for kra info and ca info to provide info on whether the ca or kra is using OAEP.
    
For the Ca, we print out oaep info for both the local CA config and the CA's corresponding KRA. This info can be used by interested clients to see if OAEP is in use with the givne KRA or CA.

### CODE REVIEW

File: base/common/src/main/java/org/dogtagpki/common/KRAInfo.java
Comment (edewata) [inline base/common/src/main/java/org/dogtagpki/common/KRAInfo.java:136]: Same question here.

File: base/server/python/pki/server/deployment/__init__.py
Comment (edewata) [reply base/server/python/pki/server/deployment/__init__.py:513]: If this code is only supposed to work in CA/KRA only, you might want to move it into `configure_ca()` and `configure_kra()` instead.

**Review by edewata — COMMENTED** [review]
I have some questions, but I think generally it looks good. I'll defer to @jmagne about the crypto aspect of it.

Would it be possible to add a key retrieval test too, maybe in a separate PR?

File: base/common/src/main/java/org/dogtagpki/common/CAInfo.java
Comment (fmarco76) [reply base/common/src/main/java/org/dogtagpki/common/CAInfo.java:141]: @edewata this is the porting of the [PR#4162](https://github.com/dogtagpki/pki/pull/4162)  where Chris raises the same problem. It can be removed without issue (I think) but it is there because the superclass implements the methods and if caller invoke the endpoint get a wrong xml. When the superclass drop the *DOM methods we can drops in all the extending classes. This was the outcome of that discussion. If you think it is better to drop them in any case we can work on that.

Comment (edewata) [reply base/common/src/main/java/org/dogtagpki/common/CAInfo.java:141]: To my understanding the `CAInfo` and `KRAInfo` are existing classes, so the XML mappings for them are already broken. Since there seems to be no complaints about them so far, I'd say anybody that needs them is using JSON, so there's no need to fix the XML mappings for these classes.

**Review by edewata — APPROVED** [review]
I'll let you decide if you want to keep the XML mappings, but it might be nice if we can drop deprecated code from the `master` branch so that eventually it can be dropped completely in the future. No need to change the other branches since they will eventually become obsolete.

Feel free to merge once @jmagne approves it. Thanks!

**Comment by fmarco76** [comment]
@edewata  I have dropped the XML and done the other change you have indicated. I'll wait @jmagne  to see if this is OK with him

**Comment by edewata** [comment]
@fmarco76 Thanks for the update!

**Review by jmagne — APPROVED** [review]
Looks good to me.

**Comment by fmarco76** [comment]
@jmagne @edewata Thanks!

**Pull request merged**
